lib/Basic/SignalHandlerWrapper.cpp

```diff
@@ -0,0 +1,44 @@
+//===- Support/SignalHandlerWrapper.cpp - Signal handling -----------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llbuild/Basic/SignalHandlerWrapper.h"
+
+#include <signal.h>
+
+using namespace llbuild;
+using namespace llbuild::basic;
+using namespace llbuild::basic::sys;
+
+SignalHandlerWrapper::SignalHandlerWrapper(Handler handler) {
+#if defined(_WIN32)
+  innerAction = handler;
+#else
+  struct sigaction action {};
+  action.sa_handler = handler;
+  innerAction = &action;
```

Comment:
That is correct, you will need to cast it to the correct type. Your initializer also isn't quite right, since it overwrites the pointer, it would need to be
```
*innerAction = action
```
but maybe that is what you meant.